lib/ex_unit/lib/ex_unit/case.ex

```diff
@@ -765,6 +765,29 @@ defmodule ExUnit.Case do
     end
   end
 
+  @doc """
+  Returns the most recently registered test case as an `%ExUnit.Test{}`
+  struct.
+
+  This is used by third-party utilities to allow compile-time configuration
+  using test tags without having to explicitly pass the test context at
+  run-time. It is indented to be invoked in macros before the test module
+  is compiled.
+
+  Raises if called with a module that has already been compiled.
+  """
+  @doc since: "1.15.0"
+  @spec most_recent_registered_test(env) :: ExUnit.Test.t() | nil
+  def most_recent_registered_test(%{module: mod}) do
+    most_recent_registered_test(mod)
+  end
+
+  def most_recent_registered_test(mod) when is_atom(mod) do
+    mod
+    |> Module.get_attribute(:ex_unit_tests)
+    |> List.first()
```

Comment:
The issue with this approach is that it will return all attributes from ETS, copying them to memory, only to get the first one. That said, I am fine with the feature proposal, but I think we need to add something like `Module.get_last_attribute` first (or get_most_recent_attribute). WDYT?

Reply:
Sounds very reasonable to me! Agreed that this is inefficient. A few follow up questions:

1. Are you thinking that `Module.get_last_attribute` would be public API or private for ExUnit/usage elsewhere within Elixir core?
2. What should it return before anything has been set? Same interface as `get_attribute`, returns nil by default and can be overridden as optional arg?
3. What should it return for non-aggregate attributes? In this case it could be equivalent to `get_attribute`, which seems fine. 
4. Would you prefer a separate PR for this or include in this PR?

Reply:
1. Public
2. Yeah, that's the most consistent
3. Yeah, same as get_attribute
4. Separate PR pls!

Reply:
Thanks for the guidance ❤️ 

Will submit a PR for the other feature and will rebase this one to use it when merged.